core/cache/namespace.go

```diff
@@ -8,31 +8,45 @@ import (
 
 	"github.com/go-logr/logr"
 	"github.com/weaveworks/weave-gitops/core/clustersmngr"
+	mngr "github.com/weaveworks/weave-gitops/core/clustersmngr"
 	v1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/rest"
 )
 
-const pollIntervalSeconds = 120
+const NamespaceStorage StorageType = "namespace"
 
 type namespaceStore struct {
-	client     clustersmngr.Client
+	logger      logr.Logger
+	restCfg     *rest.Config
+	clusterChan <-chan mngr.Cluster
+
 	namespaces map[string][]v1.Namespace
-	logger     logr.Logger
-	lock       sync.Mutex
-	cancel     func()
+
+	lock     sync.Mutex
+	cancel   func()
+	interval time.Duration
+}
+
+func WithNamespaceCache(cfg *rest.Config) WithCacheFunc {
+	return func(l logr.Logger, syncChan chan mngr.Cluster) (StorageType, Cache) {
+		return NamespaceStorage, newNamespaceStore(l, cfg, syncChan)
+	}
 }
 
-func newNamespaceStore(c clustersmngr.Client, logger logr.Logger) namespaceStore {
-	return namespaceStore{
-		client:     c,
-		namespaces: map[string][]v1.Namespace{},
-		logger:     logger,
-		lock:       sync.Mutex{},
-		cancel:     nil,
+func newNamespaceStore(logger logr.Logger, restCfg *rest.Config, clusterChan <-chan mngr.Cluster) *namespaceStore {
```

Comment:
Can we use a `NamespaceStoreOption{}` as an argument instead? That way we can add extra fields later easier, for example add option to set `interval`.

Usually after 3+ arguments I move everything into an Option.

100% not a blocker.

Reply:
no probs 👍  can add that now while I am doing the container interface tests